svc-systems/systems/storage.go

```diff
@@ -254,10 +254,108 @@ func searchItem(slice []string, val string) bool {
+		errorMessage := "Error while unmarshaling the create volume request: " + err.Error()
+		log.Printf(errorMessage)
+		resp = common.GeneralError(http.StatusBadRequest, response.MalformedJSON, errorMessage, []interface{}{}, nil)
+		return resp
+	}
+
+	// spliting the uuid and system id
+	requestData := strings.Split(req.SystemID, ":")
+	if len(requestData) != 2 || requestData[1] == "" {
+		errorMessage := "error: SystemUUID not found"
+		return common.GeneralError(http.StatusBadRequest, response.ResourceNotFound, errorMessage, []interface{}{"System", req.SystemID}, nil)
+	}
+	uuid := requestData[0]
+	target, gerr := smodel.GetTarget(uuid)
+	if gerr != nil {
+		return common.GeneralError(http.StatusBadRequest, response.ResourceNotFound, gerr.Error(), []interface{}{"System", uuid}, nil)
+	}
+	// Validating the storage instance
+	if strings.TrimSpace(req.VolumeID) == "" {
+		errorMessage := "error: Volume id is not found"
+		return common.GeneralError(http.StatusBadRequest, response.ResourceNotFound, errorMessage, []interface{}{"Volume", req.VolumeID}, nil)
+	}
+
+	// Validating the request JSON properties for case sensitive
+	invalidProperties, err := common.RequestParamsCaseValidator(req.RequestBody, volume)
+	if err != nil {
+		errMsg := "error while validating request parameters for volume creation: " + err.Error()
+		log.Println(errMsg)
+		return common.GeneralError(http.StatusInternalServerError, response.InternalError, errMsg, nil, nil)
+	} else if invalidProperties != "" {
+		errorMessage := "error: one or more properties given in the request body are not valid, ensure properties are listed in uppercamelcase "
+		log.Println(errorMessage)
+		response := common.GeneralError(http.StatusBadRequest, response.PropertyUnknown, errorMessage, []interface{}{invalidProperties}, nil)
+		return response
+	}
+
+	decryptedPasswordByte, err := e.DevicePassword(target.Password)
+	if err != nil {
+		errorMessage := "error while trying to decrypt device password: " + err.Error()
+		return common.GeneralError(http.StatusInternalServerError, response.InternalError, errorMessage, nil, nil)
+	}
+	target.Password = decryptedPasswordByte
+	// Get the Plugin info
+	plugin, gerr := smodel.GetPluginData(target.PluginID)
+	if gerr != nil {
+		errorMessage := "error while trying to get plugin details"
+		return common.GeneralError(http.StatusInternalServerError, response.InternalError, errorMessage, nil, nil)
 	}
+	var contactRequest scommon.PluginContactRequest
+	contactRequest.ContactClient = e.ContactClient
+	contactRequest.Plugin = plugin
+
+	if strings.EqualFold(plugin.PreferredAuthType, "XAuthToken") {
+		var err error
+		contactRequest.HTTPMethodType = http.MethodPost
+		contactRequest.DeviceInfo = map[string]interface{}{
+			"UserName": plugin.Username,
+			"Password": string(plugin.Password),
+		}
+		contactRequest.OID = "/ODIM/v1/Sessions"
+		_, token, getResponse, err := scommon.ContactPlugin(contactRequest, "error while creating session with the plugin: ")
```

Comment:
mocking plugin status scenario to be handled

Reply:
Handled

Reply:
handled